testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/EEConcurrencyExecutorShutdownTestCase.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    /**
+     * Tests if the server with running ConcurrencyExecutor can be stopped using cli command
+     * /host=master/server-config=server-one:stop(timeout=0)
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testConcurrencyExecutorShutdown() throws Exception {
+        ModelNode content = new ModelNode();
+        content.get("archive").set(true);
+        content.get("path").set(new File(tmpDir, "archives/" + ARCHIVE_FILE_NAME).getAbsolutePath());
+        ModelNode deploymentOpMain = createDeploymentOperation(content, MAIN_SERVER_GROUP_DEPLOYMENT_ADDRESS);
+        executeForResult(deploymentOpMain, masterClient);
+
+        this.stopServer(FIRST_SERVER_NAME);
+    }
+
+    private ModelNode createDeploymentOperation(ModelNode content, PathAddress... serverGroupAddressses) {
+        ModelNode composite = createEmptyOperation(COMPOSITE, PathAddress.EMPTY_ADDRESS);
+        ModelNode steps = composite.get(STEPS);
+        ModelNode step1 = steps.add();
+        step1.set(createAddOperation(ROOT_DEPLOYMENT_ADDRESS));
+        step1.get(CONTENT).add(content);
+        for (PathAddress serverGroup : serverGroupAddressses) {
+            ModelNode sg = steps.add();
+            sg.set(createAddOperation(serverGroup));
+            sg.get(ENABLED).set(true);
+        }
+        return composite;
+    }
+
+    /**
+     * Stops the given server and waits until STOPPED status
+     *
+     * @param serverName
+     * @throws Exception
+     */
+    private void stopServer(final String serverName) {
+        ModelNode op = new ModelNode();
+        op.get(OP_ADDR).add(HOST, MASTER);
+        op.get(OP_ADDR).add(SERVER_CONFIG, FIRST_SERVER_NAME);
+        op.get(OP).set(STOP);
+        op.get(TIMEOUT).set(0);
+        op.get(ADMIN_ONLY).set(false);
+        domainMasterLifecycleUtil.executeForResult(op);
+        try {
+            waitUntilState(FIRST_SERVER_NAME, "STOPPED");
+        } catch (TimeoutException e) {
+            fail("After \"stop(timeout=0)\" was called sever never reached the desired STOPPED state.");
+        }
+    }
+
+    /**
+     * Checks the status of he given server until given state is reached.
+     *
+     * @param serverName
+     * @param state
+     * @throws TimeoutException
+     */
+    private static void waitUntilState(final String serverName, final String state) throws TimeoutException {
+        RetryTaskExecutor<Void> taskExecutor = new RetryTaskExecutor<Void>();
+        taskExecutor.retryTask(new Callable<Void>() {
+            public Void call() throws Exception {
+                String serverStatus = this.checkServerStatus();
+                if (!serverStatus.equals(state)) throw new Exception("Server not in state " + state);
+                return null;
+            }
+
+            private String checkServerStatus() {
+                ModelNode op = new ModelNode();
+                op.get(OP_ADDR).add(HOST, "master");
```

Comment:
You have already imported the MASTER constant, you could use it here instead of "master" string.